Automattic/wp-calypso
# Editor: Redux state behavior

Related: #2994
Related: #3182

This pull request seeks to explore the second of two proposed mechanisms for managing post editor state in the Redux global state tree. The basis for this approach is that there is no state responsible for tracking the "current" edited post, and instead the assumption is that edits will always pass the desired site ID, post ID pairing to identify a post to be edited.

__Pseudo-code:__

```js
editPost( {
	title: 'Hello World'
}, 2916284, 417 );
```

__Advantages:__

- Action creators are more pure than in #3182
- No need to "start editing" a post before making revisions
 - Related: Other sections can more easily take advantage of these actions

__Disadvantages:__

- More burden on developer to determine site ID, post ID pairing in calling edit

__Open questions:__

~~Should we index edits by global ID as we do in the posts subtree? How would this work with (a) allowing multiple unsaved drafts to exist simultaneously (one-per-site?), and (b) editing posts which have yet to load (since global ID can only be determined for existing posts after post has loaded).~~
 - Decided not to use global ID

__Testing instructions:__

This has no effect on the Calypso application, because the editor reducer is not yet included in the global app state. Instead, ensure that Mocha tests pass by running `make test` from the project root directory or directly from `client/state/editor`.
